publication-event-handlers/src/main/java/no/unit/nva/publication/events/handlers/tickets/PendingPublishingRequestEventHandler.java

```diff
@@ -43,38 +41,42 @@
 
 public class PendingPublishingRequestEventHandler
     extends DestinationsEventBridgeEventHandler<EventReference, Void> {
-    
+
+    public static final String BACKEND_CLIENT_AUTH_URL = ENVIRONMENT.readEnv("BACKEND_CLIENT_AUTH_URL");
+    public static final String BACKEND_CLIENT_SECRET_NAME = ENVIRONMENT.readEnv("BACKEND_CLIENT_SECRET_NAME");
     private static final Logger logger = LoggerFactory.getLogger(PendingPublishingRequestEventHandler.class);
     private final S3Driver s3Driver;
-    private final TicketService ticketService;
     private final HttpClient httpClient;
     private final SecretsReader secretsReader;
-    public static final String BACKEND_CLIENT_AUTH_URL = ENVIRONMENT.readEnv("BACKEND_CLIENT_AUTH_URL");
-    public static final String BACKEND_CLIENT_SECRET_NAME = ENVIRONMENT.readEnv("BACKEND_CLIENT_SECRET_NAME");
     private final ResourceService resourceService;
-    
+
     @JacocoGenerated
     public PendingPublishingRequestEventHandler() {
         this(ResourceService.defaultService(),
-             defaultPublishingRequestService(),
              HttpClient.newHttpClient(),
              new SecretsReader(),
              DEFAULT_S3_CLIENT);
     }
-    
+
     protected PendingPublishingRequestEventHandler(ResourceService resourceService,
-                                                   TicketService ticketService,
                                                    HttpClient httpClient,
                                                    SecretsReader secretsReader,
                                                    S3Client s3Client) {
         super(EventReference.class);
         this.resourceService = resourceService;
-        this.ticketService = ticketService;
         this.httpClient = httpClient;
         this.secretsReader = secretsReader;
         this.s3Driver = new S3Driver(s3Client, PublicationEventsConfig.EVENTS_BUCKET);
     }
-    
+
+    protected static CognitoCredentials fetchCredentials(SecretsReader secretsReader) {
```

Comment:
Kan ikke AuthorizedBackendUriRetriever benyttes istedet for å gjenta henting av credentials etc?

Reply:
Det kan den sikkert